Indotalent.Tests/Infrastructures/Repositories/RepositoryCustomersTest.cs

```diff
@@ -0,0 +1,234 @@
+            _auditColumnTransformerMock.Object);
+
+        var exception = ThrowsAsync<Exception>(async () => await repository.GetByRowGuidAsync(null));
+        That(exception, Is.Not.Null);
+        That(exception?.Message, Is.EqualTo(ExceptionMessageRowGuidNull));
+    }
+
+    #endregion
+
+    #region TestUpdateCustomers
+
+    [TestCaseSource(typeof(Customers), nameof(Customers.GetAll))]
+    public async Task ShouldUpdateACustomer<T>(T entity) where T : _Base
+    {
+        ConfigureDbContextMock<T>();
+        var previousDateTime = entity.CreatedAtUtc;
+
+        var repository = new Repository<T>(_contextMock.Object, _httpContextAccessorMock.Object,
+            _auditColumnTransformerMock.Object);
+
+        await repository.AddAsync(entity);
+        var customer = await repository.GetByIdAsync(entity.Id);
+        That(customer, Is.Not.Null);
+
+        customer!.CreatedAtUtc = DateTime.Now;
+        await repository.UpdateAsync(customer);
+
+        _contextMock.Verify(x => x.Set<T>().Update(It.IsAny<T>()), Times.Once);
+        _contextMock.Verify(x => x.SaveChangesAsync(It.IsAny<CancellationToken>()), Times.Exactly(2));
+        Multiple(() =>
+        {
+            That(customer.CreatedAtUtc, Is.Not.EqualTo(previousDateTime));
+            That(customer.UpdatedAtUtc, Is.Not.Null);
+        });
+    }
+
+    [TestCaseSource(typeof(Customers), nameof(Customers.GetAll))]
+    public void ShouldThrowExceptionTryToUpdateNullEntity<T>(T entity) where T : _Base
+    {
+        ConfigureDbContextMock<T>();
+        var repository = new Repository<T>(_contextMock.Object, _httpContextAccessorMock.Object,
+            _auditColumnTransformerMock.Object);
+
+        var exception = ThrowsAsync<Exception>(async () => await repository.UpdateAsync(null));
+        That(exception, Is.Not.Null);
+        That(exception?.Message, Is.EqualTo(ExceptionMessageEntityNull));
+    }
+
+    #endregion
+
+    #region TestDeleteCustomers
+
+    [TestCaseSource(typeof(Customers), nameof(Customers.GetAll))]
+    public async Task ShouldDeleteACustomerByIdAsync<T>(T entity) where T : _Base
+    {
+        ConfigureDbContextMock<T>();
+        var repository = new Repository<T>(_contextMock.Object, _httpContextAccessorMock.Object,
+            _auditColumnTransformerMock.Object);
+
+        var customer = await repository.GetByIdAsync(new Random().Next(1, 5));
+        Multiple(() =>
+        {
+            That(customer, Is.Not.Null);
+            That(customer?.IsNotDeleted, Is.True);
+        });
+
+        await repository.DeleteByIdAsync(customer?.Id);
+
+        _contextMock.Verify(x =>
+            x.SaveChangesAsync(It.IsAny<CancellationToken>()), Times.Once);
+        That(customer?.IsNotDeleted, Is.False);
+
+        customer!.IsNotDeleted = true;
+        await repository.UpdateAsync(customer);
+    }
+    #endregion
+    private void ConfigureDbContextMock<T>() where T : _Base
+    {
+        var entities = (IList<T>)Customers.GetAll();
```

Comment:
Create a Customers list in `ItemLIst.cs` and follow the next convention to configure the DbContext
```
        var list = ItemList.GetList<T>(typeof(T)).BuildMockDbSet();
        list.Setup(x => x.Add(It.IsAny<T>()))
            .Callback<T>(x => ItemList.GetList<T>(typeof(T))?.Add(x));
        _contextMock.Setup(x => x.Set<T>()).Returns(list.Object);

````